lib/github-glue.ts

```diff
@@ -100,21 +101,52 @@ export class GitHubGlue {
      */
     public async addPRCc(pullRequestURL: string, cc: string):
         Promise<void> {
-        const id = cc.match(/(<.*>)/);
+        const id = cc.match(/<(.*)>/);
 
         if (!id || id[1].match(/gitster@pobox\.com/)) {
             return;
         }
 
+        const ccLower = id[1].toLowerCase();
         const url = GitGitGadget.parsePullRequestURL(pullRequestURL);
         const pr = await this.getPRInfo(url[0], url[2]);
-        const ccNow = pr.body.match(/\r?\n\r?\n(cc:.*)/is);
+        const bodyEnd = pr.body.split(/\r?\n\s*?\r?\n/).pop();
+
+        let found = false;
+        let footerSeparator = "\r\n";
+        const noFooters = "No footers";
+
+        if (bodyEnd) try {
+            bodyEnd.trimRight().split(/\r?\n/).reverse().forEach(line => {
+                const match = line.match(/^([a-z][-a-z0-9]+):\s*(.*)$/i);
+
+                if (!match) {       // stop if not a footer
+                    throw new Error(noFooters);
+                }
+
+                footerSeparator = ""; // body already has footers
+                if (match[1].toLowerCase() === "cc") {
+                    addressparser(match[2], {flatten: true}).forEach(email => {
+                        if (ccLower === email.address.toLowerCase()) {
+                            found = true;
+                            throw new Error("Found");
+                        }
+                    });
+                }
+            });
+        } catch (e) {
+            const error = e as Error;
+            if (error.message === noFooters && footerSeparator === "") {
+                throw new Error("Footer contains unknown text");
```

Comment:
I'd rather heed Postel's Law here. If, for some reason, the body ends in a paragraph whose last line starts with a word followed by a colon (which does _not_ need to be a footer!), we want to treat this as a missing footer. For example,

```
In this patch series, we do XYZ. If you wonder
why: Because ABC.
```

does _not_ have any footers.

Reply:
Okay.  Was not sure which way to go.  Was thinking the error might catch someone who failed to add the empty line but now thinking this is not the place for that.